lib/utils/getSelectorAST.mjs

```diff
@@ -0,0 +1,19 @@
+import selectorParser from 'postcss-selector-parser';
+
+/**
+ * @param {string} selector
+ * @param {import('stylelint').PostcssResult} result
+ * @param {import('postcss').Node} node
+ * @returns {import('postcss-selector-parser').Root | undefined}
+ */
+export default function getSelectorAST(selector, result, node) {
```

Comment:
[suggest] Nice-to-have. This new function `getSelectorAST` is almost the same as `parseSelector`, so I feel the former will be replaced with the latter. For example:

```diff
-import parseSelector from '../../utils/parseSelector.mjs';
+import { parseSelectorAST } from '../../utils/parseSelector.mjs';

-parseSelector(resolvedSelector, result, ruleNode, (s) => {
+parseSelectorAST(resolvedSelector, result, ruleNode)?.each((s) => {
 	checkSelector(resolvedSelector, s, ruleNode, comparisonContext);
 });
```

So, how about adding this new function to `parseSelector.mjs` like this? I guess the migration is easier.

```js
// lib/utils/parseSelector.mjs

export default function parseSelector(selector, result, node, callback) {
	// ...
}

// Move from `getSelectorAST` to here.
export function parseSelectorAST(selector, result, node) {
	// ...
}
```

Reply:
> how about adding this new function to `parseSelector.mjs`

_If_ you do that, `getSelectorAST.test.mjs` will have to be renamed `parseSelector.test.mjs`.

Reply:
I wanted to make this change, but it has the side effect of changing all the `.cjs` files that use `parseSelector ` 

<img width="1012" alt="Screenshot 2024-03-19 at 22 38 50" src="https://github.com/stylelint/stylelint/assets/11521496/454b7ebf-44a7-4cf5-92e4-2c38bad4317c">


```diff
- parseSelector(resolvedSelector, result, ruleNode, (s) => checkSelector(s, ruleNode));
+ parseSelector.default(resolvedSelector, result, ruleNode, (s) => checkSelector(s, ruleNode));
```

-----

Options:
- leave as is
- separate file, but named `parseSelectorAST.mjs` and a function `parseSelectorAST`
- refactor `export default parseSelector` to `export parseSelector` in a separate change
- ...

Reply:
> separate file, but named `parseSelectorAST.mjs` and a function `parseSelectorAST`

Okay. Can you choose the option above?

Reply:
Done!